FEXCore/include/FEXCore/Utils/LogManager.h

```diff
@@ -171,5 +171,12 @@ namespace Msg {
     FEX_TRAP_EXECUTION;             \
   } while (0)
 
+#define ERROR_AND_DIE_THROW_FMT(pred, ...) \
```

Comment:
Why does it need this need macro?

We already have like 4 other ways of asserting/dieing, and none of them have distinguishable names or are documented.

Reply:
Similar reason to the comment from #4048: `Why change these lines to LOGMAN_MSG_A_FMT? IIRC that will prevent this constexpr-evaluated function from failing at compile-time in Release builds.`

I can't use `LOGMAN_MSG_A_FMT` because this would prevent constexpr-evaluated function from failing at compile-time in Release builds.
So instead of wrapping all six lines in `GenerateFamily` with if statements, I added a throwable version of `ERROR_AND_DIE_FMT` that can be predicated.

Meanwhile we can't use a basic `static_assert` to handle the constexpr-evaluated expression because clang complains about `Static assertion expression is not an integral constant expression` due to wrapping the arguments in the CPUFamily struct to have named initializers. Which makes it easier to reason about what is getting passed to which argument.

Reply:
We can get something more equivalent to `static_assert` here by using consteval:

```cpp
// Assertion helper for code evaluated at compile-time
inline consteval void consteval_assert(bool assertion) {
    if (!assertion) {
        // Trigger compile error
        throw "consteval_assert fail";
    }
}
```

We could also add proper message support by wrapping this in a macro, but at least for the assertions in this PR we're not losing that much information as-is.